### PR TITLE
Add latency and message metrics to router control channels

### PR DESCRIPTION
**Pull request by plorenz**

### CODE REVIEW

File: controller/handler_ctrl/bind.go [hunk 1]
```diff
@@ -33,33 +35,37 @@ func newBindHandler(router *network.Router, network *network.Network, xctrls []x
...
-	traceDispatchWrapper := trace.NewDispatchWrapper(bindHandler.network.GetEventDispatcher().Dispatch)
-	ch.SetLogicalName(bindHandler.router.Id)
-	ch.AddReceiveHandler(newSessionRequestHandler(bindHandler.router, bindHandler.network))
-	ch.AddReceiveHandler(newCreateTerminatorHandler(bindHandler.network, bindHandler.router))
-	ch.AddReceiveHandler(newRemoveTerminatorHandler(bindHandler.network))
-	ch.AddReceiveHandler(newUpdateTerminatorHandler(bindHandler.network))
-	ch.AddReceiveHandler(newLinkHandler(bindHandler.router, bindHandler.network))
-	ch.AddReceiveHandler(newFaultHandler(bindHandler.router, bindHandler.network))
-	ch.AddReceiveHandler(newMetricsHandler(bindHandler.network))
+func (self *bindHandler) BindChannel(ch channel2.Channel) error {
```
Comment (michaelquigley) [inline controller/handler_ctrl/bind.go:38]: Always warms my heart.

**Review by michaelquigley — APPROVED** [review]
Looks pretty reasonable so far.

As a next step, it might also be nice if the controller had latency metrics for the control channels of all of the routers, directly. That would measure how long it takes for a message to transit from the controller to the router over the control channel, and then receive attention at the `ReceiveHandler`.

**Comment by plorenz** [comment]
> Looks pretty reasonable so far.
> 
> As a next step, it might also be nice if the controller had latency metrics for the control channels of all of the routers, directly. That would measure how long it takes for a message to transit from the controller to the router over the control channel, and then receive attention at the `ReceiveHandler`.

Looking at this again, I think I'm going to refactor this so that we have a latency response receiver and a single go-routine for sending all the latency messages. That way if the router doesn't support sending responses we won't be waiting around for something that never comes. 

By using ctrl channel specific logic we will also get direct access to the results and can hand them off to whatever is interested.

**Comment by plorenz** [comment]
> > Looks pretty reasonable so far.
> > As a next step, it might also be nice if the controller had latency metrics for the control channels of all of the routers, directly. That would measure how long it takes for a message to transit from the controller to the router over the control channel, and then receive attention at the `ReceiveHandler`.
> 
> Looking at this again, I think I'm going to refactor this so that we have a latency response receiver and a single go-routine for sending all the latency messages. That way if the router doesn't support sending responses we won't be waiting around for something that never comes.
> 
> By using ctrl channel specific logic we will also get direct access to the results and can hand them off to whatever is interested.

Added some foundation logic which allows more async style latency send/receive and also a way to plug into generic latency handler, so we can use that directly if we so desire. See: https://github.com/openziti/foundation/pull/177

**Pull request merged**